piksi_tools/console/update_downloader.py

```diff
@@ -21,10 +20,14 @@
 
 class UpdateDownloader:
     def __init__(self, root_dir=''):
-        f = urlopen(INDEX_URL)
-        self.index = jsonload(f)
-        self.root_dir = ''
-        f.close()
+        try:
+            f = requests.get(INDEX_URL)
+            self.index = f.json()
+            f.raise_for_status()
+            self.root_dir = ''
+            f.close()
+        except requests.ConnectionError as ce:
```

Comment:
this was required for no internet connection it may be we need to add in other exceptions when this fails.